src/main/java/org/cryptomator/frontend/fuse/ReadOnlyDirectoryHandler.java

```diff
@@ -31,7 +32,12 @@ public ReadOnlyDirectoryHandler(FileAttributesUtil attrUtil) {
 	}
 
 	public int getattr(Path path, BasicFileAttributes attrs, FileStat stat) {
-		stat.st_mode.set(FileStat.S_IFDIR | 0555);
+		if (attrs instanceof PosixFileAttributes) {
+			long mode = attrUtil.posixPermissionsToOctalMode(((PosixFileAttributes) attrs).permissions());
+			stat.st_mode.set(FileStat.S_IFDIR | mode);
```

Comment:
we need to remove any write permissions from the mode reported by the underlying file system

Reply:
I assume the same holds for the readonly fileHandler and linkHandler ?

Reply:
Indeed